tests/system/common.py

```diff
@@ -709,3 +709,17 @@ def deployment_wait(timeout=120, service_id=None):
         the dcos-cli and remove this method later.
     """
     shakedown.time_wait(lambda: deployment_predicate(service_id), timeout)
+
+def get_marathon_leader():
+    client = marathon.create_client()
+    about = client.get_about()
+    return about.get("leader")
+
+@retrying.retry(wait_fixed=1000, stop_max_attempt_number=30, retry_on_exception=ignore_exception)
+def assert_marathon_leadership_changed(original_leader):
```

Comment:
querying marathon for the information is interesting. because this is for a marathon test that's probably trying to test whether marathon is doing the right thing, but it's also *relying* on marathon to do the right thing (and report it as such in the /v2/info endpoint). in other words, if marathon went split-brain, how would your test know that /v2/info was reliable?

using DNS might have been flaky (because of stale data) but at least forced the test to query a 3rd party to validate who the leader was. there's value in asking a 3rd party because if there's a split-brain bug in marathon then it might not be generating reliable information in /v2/info.

an alternative 3rd party source of leader information would be the mesos v1 master/operator API (see GET_FRAMEWORKS) https://github.com/apache/mesos/blob/7beea9cbe083caf738fb02b9d58dadf3da900f4b/include/mesos/v1/master/master.proto#L61

Reply:
Interesting argument. However I would argue that the opposite it true. In case of split-brain zookeeper would:

> After the partition, all the servers in the minority region will get shutdown and moves to leader election phase. All the client sessions connected to these servers will be disconnected and will receive “KeeperState.Disconnected” event to their watchers, if any registered. A leader election will happen in both partitions and minority partition won’t be able to elect a leader. Therefore, will shutdown. Majority partition will elect a new leader and continue operation.

Hence marathon, connected to a minority partition should also shutdown, restart and won't be able to operate because there is no leader. Hence `v2/info` will not contain any leader information. 

I would argue that relying on `v2/info` is much closer to what "marathon should do", accurately mirrors what should happen during a network partition and doesn't bring in "laggy dependencies" like MesosDNS.